app/models/proxy.rb

```diff
@@ -584,23 +587,19 @@ def policies_normalize(parsed_policies)
     end
 
     def policies_configs_are_correct
-      # TODO: 5: errors.merge!(policy_config.errors)
-      select(&:invalid?).map(&:errors).each do |policy_errors|
-        policy_errors.each do |attribute, message|
-          errors.add(attribute, errors.full_message(attribute, message).downcase)
-        end
-      end
-      errors.add(:base, :missing_apicast) unless detect(&:default?)
+      errors.add(:policies_config, :invalid_policy) if any?(&:invalid?)
+      errors.add(:policies_config, :missing_apicast) unless detect(&:default?)
+      errors.add(:policies_config, :too_long, count: MAX_LENGTH) if to_s.size > MAX_LENGTH
```

Comment:
isn't it better to use standard lenght validation than adding a custom one? Actually this would count the policies config, not the resulting string length? Or am I mistaken?

Reply:
`:policies_config` is not an string in this model, it's an `Attributes::PoliciesConfig`

https://github.com/3scale/porta/blob/2cc685d34cad5c4addb033f28258264155c5a71c/app/models/proxy.rb#L19

The somehow breaks the standard rails length validation, something like `validates :policies_config, length: { maximum: MAX_LENGTH }` doesn't work.

`if to_s.size > MAX_LENGTH` count the characters after serializing the policy chain to a JSON, which is what is inserted into the proxy configuration to be pulled by apicast.

Reply:
hmm, I wonder if we better check the database serialized value length. I think
```rb
  Proxy.type_for_attribute('policies_config').serialize(self).size > MAX_LENGTH
```

Reply:
I think it's the same. I'm calling `to_s` which is an alias for `Proxy::PoliciesConfig#to_json`. On the other hand `Proxy.type_for_attribute('policies_config').serialize(self)` [also calls ](https://github.com/3scale/porta/blob/master/app/models/attributes/policies_config.rb#L9) `Proxy::PoliciesConfig#to_json`

Reply:
It is the same now. But if we use `serialize` then it will always be what is done for database. Even if we change `PoliciesConfig`.

Reply:
OK, I pushed the changes.